hypothesis-python/tests/conjecture/test_local_constants.py

```diff
@@ -0,0 +1,53 @@
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis/
+#
+# Copyright the Hypothesis Authors.
+# Individual contributors are listed in AUTHORS.rst and the git log.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+import math
+
+import pytest
+
+from hypothesis import settings, strategies as st
+from hypothesis.internal.conjecture import providers
+from hypothesis.internal.conjecture.choice import choice_equal
+
+from tests.common.debug import find_any
+from tests.common.utils import Why, xfail_on_crosshair
+
+
+# I tried using @given(st.integers()) here, but I think there is a bad interaction
+# with CONSTANTS_CACHE when testing it inside of a hypothesis test.
+@pytest.mark.parametrize("value", [2**20 - 50, 2**10 - 10, 129387123, -19827321, 0])
+def test_can_draw_local_constants_integers(monkeypatch, value):
+    monkeypatch.setattr(providers, "local_constants", lambda: {"integer": {value}})
+    find_any(st.integers(), lambda v: choice_equal(v, value))
+
+
+@xfail_on_crosshair(Why.undiscovered)  # exact float match is hard for smt
+@pytest.mark.parametrize("value", [1.2938, -1823.0239, 1e999, math.nan])
+def test_can_draw_local_constants_floats(monkeypatch, value):
+    monkeypatch.setattr(providers, "local_constants", lambda: {"float": {value}})
+    find_any(st.floats(), lambda v: choice_equal(v, value))
```

Comment:
Exact float match should be trivial; the hard part is that we're converting the floats into integers via `struct.pack` before doing the comparison.  Maybe that's still a Crosshair issue for this specific case though?

Either way we should update the comment.

Reply:
I assumed exact match was hard because crosshair mimics fp64 semantics with many symbolics, but you're probably right that it's struct.pack

FWIW it only couldn't find one of them (I think `1.2938`)